### ISSUE TITLE
Lint suggestion: Mismatched macro repetition operator

### ISSUE THREAD

**Issue by theemathas** [rust-lang/rust-clippy#16138]
### What it does

In a `macro_rules!` definition, there are three different [repetition operators](https://doc.rust-lang.org/stable/reference/macros-by-example.html#r-macro.decl.repetition) that can be used: `*`, `+`, `?`. Rust accepts the code even if the repetition operator in the macro matcher is different from the repetition operator in the macro transcriber. This lint would warn against such a mismatch.

### Advantage

- Having mismatched repetition operators can be confusing, since the kind of repetition operator used in the macro transcriber is ignored, and therefore can be misleading.
- Mismatched repetition operators can be a sign of a bug, where the macro author intended to refer to a different "repetition group".

### Example

```rust
macro_rules! my_macro {
    ($($i:ident)?) => {
        $($i)*
    }
}
```

Could be written as:

```rust
macro_rules! my_macro {
    ($($i:ident)?) => {
        $($i)?
    }
}
```

**Comment by PatchMixolydic** [comment]
For what it's worth, [`meta_variable_misuse`](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#meta-variable-misuse), an allow-by-default `rustc` lint, already catches this ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=a5772ddb902ce1f3b5eb0ac34255a4da)):

```
warning: meta-variable repeats with different Kleene operator
 --> src/lib.rs:5:11
  |
4 |     ($($i:ident)?) => {
  |                 - expected repetition
5 |         $($i)*
  |           ^^ - conflicting repetition
  |
note: the lint level is defined here
 --> src/lib.rs:1:9
  |
1 | #![warn(meta_variable_misuse)]
  |         ^^^^^^^^^^^^^^^^^^^^
```
